src/main/kotlin/com/openlattice/organization/DatasetApi.kt

```diff
@@ -0,0 +1,167 @@
+package com.openlattice.organization
+
+import retrofit2.http.*
+import java.util.*
+
+interface DatasetApi {
+
+    companion object {
+        /* These determine the service routing */
+        const val SERVICE = "/datastore"
+        const val CONTROLLER = "/organization-database"
+        const val BASE = SERVICE + CONTROLLER
+
+        const val EXTERNAL_DATABASE = "/external-database"
+        const val EXTERNAL_DATABASE_COLUMN = "/external-database-column"
+        const val EXTERNAL_DATABASE_TABLE = "/external-database-table"
+        const val DATA = "/data"
+
+        const val ID = "id"
+        const val ID_PATH = "/{$ID}"
+        const val TABLE_NAME = "tableName"
+        const val TABLE_NAME_PATH = "/{$TABLE_NAME}"
+        const val TABLE_ID = "tableID"
+        const val TABLE_ID_PATH = "/{$TABLE_ID}"
+        const val COLUMN_NAME = "columnName"
+        const val COLUMN_NAME_PATH = "/{$COLUMN_NAME}"
+        const val ROW_COUNT = "rowCount"
+        const val ROW_COUNT_PATH = "/{$ROW_COUNT}"
+        const val USER_ID = "userId"
+        const val USER_ID_PATH = "/{$USER_ID}"
+        const val CONNECTION_TYPE = "connectionType"
+        const val CONNECTION_TYPE_PATH = "/{$CONNECTION_TYPE}"
+    }
+
+    /**
+     *
+     * @param organizationId The organization's UUID
+     * @param userId The user to be given trusted status
+     * @param connectionType The connection type via which the user will be trusted
+     * @param ipAddresses The set of ip addresses (as strings) from which users
+     * will be trusted. This should be an empty set if the connectionType is local.
+     */
+    @POST(BASE + ID_PATH + USER_ID_PATH + CONNECTION_TYPE_PATH + EXTERNAL_DATABASE)
+    fun addHBARecord(
+            @Path(ID) organizationId: UUID,
+            @Path(USER_ID) userId: String,
+            @Path(CONNECTION_TYPE) connectionType: String,
+            @Body ipAddresses: Set<String>)
+
+    @DELETE(BASE + ID_PATH + USER_ID_PATH + EXTERNAL_DATABASE)
+    fun removeHBARecord(
+            @Path(ID) organizationId: UUID,
+            @Path(USER_ID) userId: String)
+
+    //get
+
+    /**
+     * Gets all OrganizationExternalDatabaseTable objects for an organizatoin
+     * @param organizationId The organization's UUID
+     */
+    @GET(BASE + ID_PATH + EXTERNAL_DATABASE_TABLE)
+    fun getExternalDatabaseTables(@Path(ID) organizationId: UUID): Set<OrganizationExternalDatabaseTable>
+
+    /**
+     * Gets all OrganizationExternalDatabaseTable objects mapped to
+     * their containing OrganizationExternalDatabase columns
```

Comment:
not super important but the wording here is a bit confusing

Reply:
changed it, hopefully it is better